### PR TITLE
(#1460) Introduce Flattened and use it where relevant

### PR DESCRIPTION
**Pull request by victornoel**
This is for #1460.

### CODE REVIEW

File: src/main/java/org/cactoos/text/Abbreviated.java [hunk 1]
```diff
@@ -80,32 +82,24 @@ public Abbreviated(final String text, final int max) {
...
-                            ).asString(),
-                            Abbreviated.ELLIPSES
-                        );
-                    }
-                    return abbreviated.asString();
-                }
+            new Flattened(
+                new Ternary<>(
+                    new ScalarOf<>(() -> new Sticky(text)),
+                    (Text t) -> t.asString().length() <= max,
```
Comment (andreoss) [inline src/main/java/org/cactoos/text/Abbreviated.java:91]: @victornoel `LengthOf` could be used here.

**Review by andreoss — CHANGES_REQUESTED** [review]

File: src/main/java/org/cactoos/text/Abbreviated.java (hunk 1 above)
Comment (victornoel) [reply src/main/java/org/cactoos/text/Abbreviated.java]: @andreoss done, thx

**Comment by victornoel** [comment]
@andreoss see the corrections + the new commit :)

**Comment by victornoel** [comment]
@andreoss is it ok?

**Comment by victornoel** [comment]
@andreoss can we merge this?

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#1515](https://github.com/yegor256/cactoos/issues/1515) is not in the agenda of @andreoss[/z](https://www.0crat.com/u/andreoss), can't set title

**Comment by 0crat** [comment]
Job [#1515](https://github.com/yegor256/cactoos/issues/1515) is not in the agenda of @andreoss[/z](https://www.0crat.com/u/andreoss), can't set title

**Comment by 0crat** [comment]
Code review was too long (13 days), architects (@victornoel) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/yegor256/cactoos/issues/1515#issuecomment-758460355))

@sereshqua There is no quality review for [#1515](https://github.com/yegor256/cactoos/issues/1515), no performer

**Pull request merged**